lib/matplotlib/tests/test_widgets.py

```diff
@@ -1029,6 +1044,29 @@ def test_check_buttons(fig_test, fig_ref):
     ax.text(.25, 1/3, "coffee", transform=ax.transAxes, va="center")
 
 
+@check_figures_equal(extensions=['png'])
+def test_check_button_props(fig_test, fig_ref):
+    label_props = {'color': ['red'], 'fontsize': [24]}
+    frame_props = {'facecolor': 'green', 'edgecolor': 'blue', 'linewidth': 2}
+    check_props = {'facecolor': 'red', 'linewidth': 2}
+
+    widgets.CheckButtons(fig_ref.subplots(), ['tea', 'coffee'], [True, True],
+                         label_props=label_props, frame_props=frame_props,
+                         check_props=check_props)
+
+    cb = widgets.CheckButtons(fig_test.subplots(), ['tea', 'coffee'],
+                              [True, True])
+    cb.set_label_props(label_props)
+    # Setting the label size automatically increases default marker size, so we
+    # need to do that here as well.
+    cb.set_frame_props({**frame_props, 's': (24 / 2)**2})
+    # FIXME: Axes.scatter promotes facecolor to edgecolor on unfilled markers,
+    # but Collection.update doesn't do that (it forgot the marker already).
+    # This means we cannot pass facecolor to both setters directly.
+    check_props['edgecolor'] = check_props.pop('facecolor')
```

Comment:
This is my only hangup for API consistency right now. On unfilled markers, `Axes.scatter` will promote `facecolor` to `edgecolor`. But `Collection` does not do that, and in fact, it has _forgotten_ about the marker entirely. We can handle this in `set_check_props`, but it would require some internal bookkeeping of the configured `marker`.

Alternatively, we could say it's not allowed to change the 'check' marker style, and only accept `color` that we internally translate to the right argument.